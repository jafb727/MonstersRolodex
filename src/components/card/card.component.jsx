/**
 * Filename: card.component.js
 * Author: Jose A Felix
 * Description: Card component
 */

// ----------------------------------------------------------------
/** Components */

import React from "react";

// ----------------------------------------------------------------
/** Stylesheet */

//import "./card.styles.css";

// ----------------------------------------------------------------

// Main component
export const Card = ({ id, name, email }) => {
	return (
		<div className="card-container">
			<img
				alt="monster"
				src={`https://robohash.org/${id}?set=set2&size=180x180`}
			/>
			<h2>{name}</h2>
			<p>{email}</p>
			<p>{email}</p>
		</div>
	);
};
