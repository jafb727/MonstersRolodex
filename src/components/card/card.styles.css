/**
 * Filename: card.styles.css
 * Author: Jose A Felix
 * Description: Card component style
 */

 /** ---------------------------------------------------------------- */

 .card-container{
      display:flex;
      align-items: center;
      flex-direction: column;
      border-radius:0.5rem;
      padding:1.5rem;
      cursor: pointer;
     -moz-osx-font-smoothing:grayscale;
     backface-visibility: hidden;
     transform: translateZ(0);
     transition: transform 0.25s ease-out;
     background: linear-gradient(#dc0c98, #3d076d);
 }

 .card-container:hover{
     transform: scale(1.05);
     background: #ff0099;
}

 .card-container img{
      border-radius: 50%;
      max-width: 70%;
      border:5px solid #490862;
 }

 .card-container h2{
     font-size: 2.5rem;
     font-family: "Bigelow Rules";
     margin-bottom: 0;
     color:#28ffe3;
     text-shadow:3px 3px 0 #006661
 }

 .card-container p{
      font-family: "Segoe UI Light";
      font-size: 1rem;
      color: #71ffec;
 }

