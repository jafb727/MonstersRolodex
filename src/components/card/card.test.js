import { shallow } from "enzyme";
import React from "react";
import { Card } from "./card.component";

it("Testing Card component", () => {
	expect(shallow(<Card />)).toMatchSnapshot();
});
