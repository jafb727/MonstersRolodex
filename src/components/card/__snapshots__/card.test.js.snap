// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card component 1`] = `
<div
  className="card-container"
>
  <img
    alt="monster"
    src="https://robohash.org/undefined?set=set2&size=180x180"
  />
  <h2 />
  <p />
  <p />
</div>
`;
